src/SeriesProvider/SeriesProvider.py

```
@@ -105,38 +90,89 @@ def request_output(self, semaphoreSeriesDescription: SemaphoreSeriesDescription,
+        """This function validates a set of results against a request. It will generate a list of every expected time stamp and 
+        attempt to match inputs to those timestamps. If both a database and data ingestion series are provided it will merge them
+        prioritizing results from data ingestion when they conflict with results from the database.
+        :param seriesDescription: SemaphoreSeriesDescription - The request description
+        :param timeDescription: TimeDescription - The description of the temporal information of the request 
+        :param DBSeries: Series - A series of results to validate from the DB
+        :param DISeries: Series - A series of results to validate from data ingestion
+        :return a validated series: Series
 
-    def __merge_results(self, first: list, second: list) -> list:
-        """ Merges two lists together, will only keep unique results.
-            :param first: list[any] - The first list.
-            :param second: list[any] - The second list.
-            :return list: The combined lists.
         """
+        missing_results = 0
+        datetimeDict = self.__generate_datetime_map(timeDescription) # A dict with every time stamp we expect, with a value of none
 
-        #TODO:: This is a very slow, it was optimized with hashing like this:    
-            # uniqueToSecond = set(second) - set(first)
-            # return first + list(uniqueToSecond)
-        #But its not hashing properly anymore, needs to be looked into
+        # Iterate over every timestamp we are expecting 
+        for datetime in datetimeDict.keys():
+            
+            # Search the DB results for every input that matches that timestamp
+            valid_database_results = [input for input in DBSeries.data if input.timeGenerated == datetime]
+            if len(valid_database_results) > 1:
+                log(f'Warning: DB results returned multiple results for the same datetime \n{DBSeries}\n{valid_database_results}')
```

are there scenarios where this could happen?  I think we need to make sure that this should not happen, and if it does, this should constitute an error and we should throw an error rather than just logging.  When we ask the persist layer to retrieve our inputs, it should do so by always getting the ones with the most recent time acquired timestamp.

---

Solved by switching to dicts